.github/workflows/zinc.yaml

```diff
@@ -0,0 +1,56 @@
+name: Zinc Charm Integration Tests
+
+on: [push, pull_request]
+
+jobs:
+  zinc:
+    name: Zinc Charm Integration Tests
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out python-libjuju
+      uses: actions/checkout@v3
+      with:
+        path: python-libjuju
+
+    - name: Check out zinc-k8s-operator
+      uses: actions/checkout@v3
+      with:
+        repository: jnsgruk/zinc-k8s-operator
+        path: zinc-k8s-operator
+
+    - name: Setup Microk8s
+      uses: balchua/microk8s-actions@v0.3.0
+      with:
+        channel: 1.25-strict/stable
+        addons: '["dns", "ingress", "hostpath-storage"]'
+
+    - name: Install dependencies
+      run: |
+        sudo apt-get remove lxd lxd-client
+        if snap info lxd | grep "installed"; then
+          sudo snap refresh lxd --channel=latest/stable
+        else
+          sudo snap install lxd --channel=latest/stable
+        fi
+
+        sudo lxd waitready
+        sudo lxd init --auto
+        sudo chmod a+wr /var/snap/lxd/common/lxd/unix.socket
+        sudo usermod -a -G lxd $USER
+
+        # Consider changing to latest/stable once that channel catches up to 3.0
+        sudo snap install juju --channel=3.0/stable
+        mkdir -p ~/.local/share
+        # sudo snap install charmcraft --classic
+
+    - name: Bootstrap Juju
+      run: sg snap_microk8s "juju bootstrap microk8s microk8s"
```

Comment:
I'd seriously consider replacing this boilerplate with the `actions-operator` that the charm teams use, which takes care of all of this setup for you: https://github.com/jnsgruk/zinc-k8s-operator/blob/0ef6a80bd3d89fc4d34aca1a57fabfc099443919/.github/workflows/build-and-test.yaml#L56-L62